diff -U 3 -r mysql-gui-tools-5.0r12-orig/mysql-gui-common/configure.in mysql-gui-tools-5.0r12/mysql-gui-common/configure.in
--- mysql-gui-tools-5.0r12-orig/mysql-gui-common/configure.in	2007-03-29 17:46:22.000000000 +0300
+++ mysql-gui-tools-5.0r12/mysql-gui-common/configure.in	2007-12-13 21:23:58.000000000 +0200
@@ -336,7 +336,9 @@
   if (pcre_compile("(?:\\\\pL(?:\\\\pL|\\\\pN|\\\\pM|\\\\p{Pc}|\\\\p{Pd}|\\\\pS}|_)*)", PCRE_UTF8, &err, &o, NULL)) 
     return 0;
   return 1;
-])], [HAVE_UNICODE_PCRE=1 AC_MSG_RESULT(yes)], [HAVE_UNICODE_PCRE=0 AC_MSG_RESULT(no)])
+])], [HAVE_UNICODE_PCRE=1
+AC_MSG_RESULT(yes)], [HAVE_UNICODE_PCRE=0
+AC_MSG_RESULT(no)])
 CPPFLAGS=$ac_savecpp
 LIBS=$ac_savelibs
 AC_DEFINE(HAVE_UNICODE_PCRE, [], [whether pcre was compiled with unicode properties support])
diff -U 3 -r mysql-gui-tools-5.0r12-orig/mysql-query-browser/source/linux/MQQueryDispatcher.cc mysql-gui-tools-5.0r12/mysql-query-browser/source/linux/MQQueryDispatcher.cc
--- mysql-gui-tools-5.0r12-orig/mysql-query-browser/source/linux/MQQueryDispatcher.cc	2007-04-27 16:11:32.000000000 +0300
+++ mysql-gui-tools-5.0r12/mysql-query-browser/source/linux/MQQueryDispatcher.cc	2007-12-13 21:23:59.000000000 +0200
@@ -558,8 +558,8 @@
 
   Gtk::Main::instance()->run();
 
-  while(!req->is_complete())
-    ;
+  //while(!req->is_complete())
+  //  ;
 
   return sps;
 }
